### PR TITLE
Issue #DP-30 fix: Set hierarchy open for contributon and chapters count

### PR DESCRIPTION
**Pull request by ahghatol**

### CODE REVIEW

File: src/helpers/updateHierarchy.helper.js [hunk 1]
```diff
@@ -146,6 +147,15 @@ class HierarchyService {
     };
   }
 
+  openForContribution(data, openForContribution) {
+    let instance = this;
```
Comment (venkateshwarans) [inline src/helpers/updateHierarchy.helper.js:151]: Please do not have the same name for method and parameter

Comment (ahghatol) [reply src/helpers/updateHierarchy.helper.js:151]: fixed @venkateshwarans

File: src/helpers/programHelper.js [hunk 2]
```diff
@@ -203,14 +203,22 @@ getContributionWithProgramId(program_id, req) {
...
+
+        tableData = _.map(openForContributionCollections, (collection) => {
         const result = {};
         // sequence of columns in tableData
         result['Textbook Name'] = collection.name;
         result['Medium'] = collection.medium || '--';
         result['Class'] = collection.gradeLevel && collection.gradeLevel.length ? collection.gradeLevel.join(', ') : '';
         result['Subject'] = collection.subject || '--';
-        result['Number of Chapters'] = collection.chapterCount || '--';
+        result['Number of Chapters'] = collection.chapterCountForContribution || collection.chapterCount || '--';
```
Comment (venkateshwarans) [inline src/helpers/programHelper.js:221]: Can you check  if currently `chapterCount` is a number or string? 
And let's not keep the fallback value as `--`. it should be 0 or `0`.

Comment (venkateshwarans) [reply src/helpers/programHelper.js:221]: let's remove `--` from `medium` and `subject` etc... as well

File: src/helpers/updateHierarchy.helper.js [hunk 3]
```diff
@@ -157,6 +167,24 @@ class HierarchyService {
       isFirstTime: true
     };
 
+    // Set textbook allow to open for contribution
+    response.content['openForContribution'] = true;
+    let chapterCountForContribution = 0;
+
+    _.forEach(response.content.children, (child, i) => {
+        let cindex = children.findIndex(item => item.id === child.identifier);
+
```
Comment (venkateshwarans) [inline src/helpers/updateHierarchy.helper.js:176]: Here, what I would ideally do it, if we are not going to use the `index`, I would rather use `_.includes`. 
Here it is not about right or wrong, just the readability.

Comment (ahghatol) [reply src/helpers/updateHierarchy.helper.js:176]: closing as discussed

**Pull request merged**